Assets/Scripts/ConnectionManagement/ConnectionState/HostingState.cs

```diff
@@ -25,15 +25,17 @@ class HostingState : ConnectionState
 
         public override void Enter()
         {
-            var gameState = UnityEngine.Object.Instantiate(m_ConnectionManager.GameState);
-
-            gameState.Spawn();
+            foreach (var prefab in m_ConnectionManager.NetworkObjectsToSpawnWhenServerStarted)
+            {
+                var networkObject = UnityEngine.Object.Instantiate(prefab);
+                networkObject.Spawn();
+            }
 
             SceneLoaderWrapper.Instance.AddOnSceneEventCallback();
 
             //The "BossRoom" server always advances to CharSelect immediately on start. Different games
             //may do this differently.
-            SceneLoaderWrapper.Instance.LoadScene("CharSelect", useNetworkSceneManager: true);
+            SceneLoaderWrapper.Instance.LoadScene(m_ConnectionManager.StartScene, useNetworkSceneManager: true);
```

Comment:
See my DGS PR for this, connection management shouldn't manage scene names.

Reply:
I agree with this as well, but again I don't think it is in the scope of this PR

Reply:
If you agree my DGS PR is the way to go to manage those names vs having them in assets, could you revert this so this makes merging easier with less conflicts? This way I wouldn't have to revert your change you're introducing with this PR